### PR TITLE
refactor(app): robot overview section of device details page

### LINKED ISSUE #8691: Device Details: Robot Overview section
# Overview
This ticket covers the top section of the device details page where we display the robot name and picture, connection status, light switch, and primary and secondary actions.

# Design
This artboard contains the designs for the Robot Overview section based on the state of the robot (available+idle, available+busy, robot powering on, or robot not available): 
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A91474

# Implementation Details
- The "Run a Protocol" button should only be active when a robot is available, idle, and with at least 1 pipette attached
- The "Robot Settings" button should not appear if a robot is not available/connected
- The robot lights switch should not appear if a robot is not available/connected

# Acceptance Criteria

- [ ] The robot type, image, and name render correctly

- [ ] The robot status and network connectivity are displayed

- [ ] The user can turn on and off robot lights using the toggle

- [ ] The user can open the robot settings page for any available robot

- [ ] The user can run a protocol if the robot is available + idle

- [ ] Clicking the "Run a Protocol" button should open the "Choose a Protocol" slide-out where the user can select the protocol they'd like to run on the selected robot

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

This is the component and main functionality for the robot overview that renders on the unified app device details page:
 - Some commonality with the robot card (devices landing page) has been pulled out into a new shared component, `RobotStatusBanner`. 
 - The device details page is created as well, with a placeholder spot for module cards. 
 - The device details page can be accessed through clicking the "three dots" icon on a robot card (work to be fleshed out in robot actions menu ticket #8673)
 - Also contains some cleanup of the temp menu

Please (mostly) ignore styling as design overhaul is in progress for this page. 

closes #8691

# Changelog

 - Adds Robot Overview component to the device details page

# Review requests

 - confirm robot information is correct for active/idle runs
 - check that the lights toggle works, on physical robot ideally
 - review general page and component structure

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/RobotOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  TEXT_TRANSFORM_UPPERCASE,
+} from '@opentrons/components'
+import { ApiHostProvider } from '@opentrons/react-api-client'
+
+import { ToggleBtn } from '../../atoms/ToggleBtn'
+import { useLights, useRobot, useIsProtocolRunning } from './hooks'
+import { RobotStatusBanner } from './RobotStatusBanner'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const OT2_PNG = require('../../assets/images/OT2-R_HERO.png')
```
Comment (shlokamin) [inline app/src/organisms/Devices/RobotOverview.tsx]: can this not be imported using an `import` statement?

File: app/src/organisms/Devices/__tests__/RobotCard.test.tsx [hunk 2]
```diff
@@ -69,64 +63,9 @@ describe('RobotCard', () => {
...
     const [{ getByText }] = render()
-
-    expect(getByText('Running Testosaur')).toBeTruthy()
-
-    const runLink = getByText('Go to Run')
-    expect(runLink).toBeTruthy()
-    expect(runLink.getAttribute('href')).toEqual(
-      '/devices/otie/protocol-runs/run'
-    )
+    expect(getByText('Mock RobotStatusBanner')).toBeTruthy()
```
Comment (shlokamin) [inline app/src/organisms/Devices/__tests__/RobotCard.test.tsx]: `getBy*` queries will throw if they don't find anything, so the `.toBeTruthy()` is actually unnecessary. in know we do this all over the place (i see it was done before you even touched this code) but we should try to stop doing that IMO

File: app/src/organisms/Devices/__tests__/RobotOverview.test.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  })
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('renders an OT image', () => {
+    const [{ getByRole }] = render()
+    const image = getByRole('img')
+
+    expect(image).toBeTruthy()
```
Comment (shlokamin) [inline app/src/organisms/Devices/__tests__/RobotOverview.test.tsx]: same with all of the `toBeTruthy()` statements here

File: app/src/organisms/Devices/__tests__/RobotStatusBanner.test.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    mockUseCurrentProtocolRun.mockReturnValue({} as UseCurrentProtocolRun)
+    mockUseIsProtocolRunning.mockReturnValue(false)
+  })
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('renders the type of robot', () => {
+    const [{ getByText }] = render()
+    expect(getByText('OT-2')).toBeTruthy()
```
Comment (shlokamin) [inline app/src/organisms/Devices/__tests__/RobotStatusBanner.test.tsx]: and these too

File: app/src/organisms/Devices/hooks.ts [hunk 5]
```diff
@@ -42,3 +52,41 @@ export function useAttachedPipettes(
...
+
+  return { lightsOn, toggleLights }
+}
+
+export function useIsProtocolRunning(): boolean {
+  const { runRecord } = useCurrentProtocolRun()
+
+  // may want to adjust the condition that shows the active run - only running, paused, etc
+  const isProtocolRunning =
+    runRecord != null && runRecord.data.status !== RUN_STATUS_IDLE
```
Comment (shlokamin) [inline app/src/organisms/Devices/hooks.ts:89]: i agree with this comment. i think doing that now will save us some headaches later down the road 😄

File: app/src/pages/Devices/DeviceDetails/__tests__/DeviceDetails.test.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    const [{ queryByText }] = render('/devices/otie')
+
+    expect(queryByText('Mock RobotOverview')).toBeFalsy()
+  })
+
+  it('renders a RobotOverview when a robot is found', () => {
+    mockUseRobot.mockReturnValue(mockConnectableRobot)
+    const [{ getByText }] = render('/devices/otie')
+
+    expect(getByText('Mock RobotOverview')).toBeTruthy()
```
Comment (shlokamin) [inline app/src/pages/Devices/DeviceDetails/__tests__/DeviceDetails.test.tsx]: `toBeTruthy` isn't necessary

File: app/src/organisms/Devices/RobotOverview.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      <Flex
+        alignItems={ALIGN_CENTER}
+        backgroundColor={C_WHITE}
+        borderBottom={`1px solid ${C_MED_LIGHT_GRAY}`}
+        flexDirection={DIRECTION_ROW}
+        marginBottom={SPACING_2}
+        padding={SPACING_2}
+        width="100%"
+      >
+        <img src={OT2_PNG} width="93px" />
```
Comment (shlokamin) [inline app/src/organisms/Devices/RobotOverview.tsx]: is there a way we can avoid this width being in pixels? tbh not even sure what our standard practice is @b-cooper knows more

**Review by shlokamin — COMMENTED** [review]
exciting stuff! left some comments nothing major though. would love to see this get merged ASAP so we can kick off work on the module cards. i haven't been involved in next gen app planning though so I'll let someone else give the stamp of approval

**Review by jerader — APPROVED** [review]
Nice job, code makes sense to me. Really cool to visually see everything and the amount of detail such as modules and pipettes on the Devices and Details page

File: app/src/organisms/Devices/RobotOverview.tsx (hunk 7 above)
Comment (brenthagen) [reply app/src/organisms/Devices/RobotOverview.tsx]: our powers of two SIZE constants don't come very close to the design width. i can change this to `6rem` though, that's probably better than px

File: app/src/organisms/Devices/RobotOverview.tsx (hunk 1 above)
Comment (brenthagen) [reply app/src/organisms/Devices/RobotOverview.tsx]: `import` works, changed

File: app/src/organisms/Devices/hooks.ts (hunk 5 above)
Comment (brenthagen) [reply app/src/organisms/Devices/hooks.ts:89]: makes sense - have we defined the run statuses that correspond to an active run? offhand I'd say `RUN_STATUS_RUNNING`, `RUN_STATUS_PAUSE_REQUESTED`, `RUN_STATUS_PAUSED`, and `RUN_STATUS_STOP_REQUESTED` but do we want to also include the completed statuses? @emilywools

Comment (brenthagen) [reply app/src/organisms/Devices/hooks.ts:89]: possibly `RUN_STATUS_FINISHING` as well

**Review by shlokamin — APPROVED** [review]
🚀

**Pull request merged**